src/mlpack/methods/pca/decomposition_policies/stochastic_method.hpp

```diff
@@ -0,0 +1,323 @@
+/**
+ * @file stochastic_method.hpp
+ * @author Marcus Edel
+ *
+ * Implementation of the stochastic method for use in the Principal Components
+ * Analysis method.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_PCA_DECOMPOSITION_POLICIES_STOACHASTIC_METHOD_HPP
+#define MLPACK_METHODS_PCA_DECOMPOSITION_POLICIES_STOACHASTIC_METHOD_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/optimizers/sgd/sgd.hpp>
+#include <mlpack/core/optimizers/adam/adam.hpp>
+#include <mlpack/core/optimizers/svrg/svrg.hpp>
+
+#include <mlpack/core/util/sfinae_utility.hpp>
+
+namespace mlpack {
+namespace pca {
+
+//! Detect an StepSize() method.
+HAS_MEM_FUNC(StepSize, HasStepSize);
+
+/**
+ * Implementation of the Stochastic Policy. The stochastic approximation
+ * algorithm is an iterative algorithm, where in each iteration a single
+ * sampled point is used to perform an update as done in Stochastic Gradient
+ * Descent. For PCA this means iteratively using vectors sampled from a
+ * distribution to update the subspace.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @inproceedings{Musco2015,
```

Comment:
Maybe this should be `arora2012`?  I met one of the Musco brothers at NIPS last year, but I can't remember which one (probably Cameron?).  A fun person to talk to. :)

Reply:
You are absolutely right.